tests/integration-test/src/main.rs

```diff
@@ -29,12 +11,7 @@ fn main() -> anyhow::Result<()> {
     // Run the tests
     for t in inventory::iter::<IntegrationTest> {
         info!("Running {}", t.name);
-        debug!("Starting bpfd");
-        let mut cmd = Command::cargo_bin("bpfd")?;
-        let c = cmd.spawn()?;
-        let _guard = ChildGuard(c);
-        // let bpfd start up
-        sleep(Duration::from_secs(2));
+        let _guard = vec![start_bpfd()?];
         if let Err(e) = (t.test_fn)() {
```

Comment:
I think the best option, if you're up for it, would be to have all integration tests be:
```rust
#[integration_test]
fn some_cool_test(bpfd_guard: ChildGuard) -> anyhow::Result() {
   // test code goes here
}
```

Then stopping/starting bpfd should be fairly easy.
```rust
drop(bpfd_guard);
let _guard = start_bpfd()?;
```

Reply:
@dave-tucker I like loading it from the test as well. Not sure how to pass it as a parameter though (because I'm not Dave :sweat_smile:). Changed it to the following for the time being.

```rust
fn test_load_unload() -> anyhow::Result<()> {
    let _guard = start_bpfd()?;
```
Inside of tests the pattern would be:

```rust
    let _guard = start_bpfd()?;
    // <do something> then kill the process
    drop(_guard);
    // start the process again and make sure things still work
    let _guard = start_bpfd()?;
```